azblob/bytesreadercloser.go

```diff
@@ -0,0 +1,18 @@
+package azblob
+
+import "bytes"
+
+type BytesReaderCloser struct {
+	*bytes.Reader
+}
+
+func NewBytesReaderCloser(b []byte) *BytesReaderCloser {
+	r := &BytesReaderCloser{
+		Reader: bytes.NewReader(b),
+	}
+	return r
+}
+
+func (io *BytesReaderCloser) Close() error {
```

Comment:
There is a function io.NopCloser in sthe stdlib which does exactly this - adds a Close method that does nothing

Reply:
ah, lovely. thanks

Reply:
ugh, it needs a Seek ...

Reply:
Yeah so it needs a Seek method so I couldn't do this in the end.